### ISSUE TITLE
core: async observable update not triggering change detection in ejs-grid e-column directive.

### ISSUE THREAD

**Issue by kumaresan-subramani** [angular/angular#36503]
# 🐞 bug report

In angular 9, ngx-translate is not working as expected.

### Is this a regression?

yes, working fine in angular 8

### Description

while using ngx-translate to translate text, it return empty string as text but working fine in angular 8.

> **src\app\components\grid.component.html**

```
<ejs-grid #grid [dataSource]="items" allowPaging="true" allowSorting="true" allowMultiSorting="true" (dataStateChange)="dataStateChange($event)" [locale]="locale" [pageSettings]="pageSettings">
    <e-columns>
      <e-column field="displayValue" headerText="{{'GRID.DISPLAYVALUE' | translate}}" width="80"></e-column>
      <e-column width="30" headerText="afaf" textAlign="center">
        <ng-template #template let-data>  
          <a (click)="delete(data.keyValue)">Delete</a>
        </ng-template>
      </e-column>
    </e-columns>
  </ejs-grid>
```

## 🔬 Minimal Reproduction
1.download below sample
[ngTranslateSampleIssue-712573671.zip](https://github.com/angular/angular/files/4449447/ngTranslateSampleIssue-712573671.zip)

2. Give `npm i`
3. Then `ng serve`

you will see below in browser page. **no header text**

![image](https://user-images.githubusercontent.com/36756050/78766398-d7c89400-79a6-11ea-8f69-960e2872c504.png)

## 🔥 Exception or Error
`headertext` property has **null** value in component level hook.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
dependencies": {
    "@angular/animations": "9.0.4",
    "@angular/common": "9.0.4",
    "@angular/compiler": "9.0.4",
    "@angular/core": "9.0.4",
    "@angular/forms": "9.0.4",
    "@angular/platform-browser": "9.0.4",
    "@angular/platform-browser-dynamic": "9.0.4",
    "@angular/router": "9.0.4",
    "@ngx-translate/core": "^11.0.1",
    "@ngx-translate/http-loader": "^4.0.0",

</code></pre>

**Anything else relevant?**
my component life cycle hooks gets triggered from angular core level, in there my header-text property alue is null when i use translate like below

`<e-column field="displayValue" headerText="{{'GRID.DISPLAYVALUE' | translate}}" width="80"></e-column>`

### Refer below image

![image](https://user-images.githubusercontent.com/36756050/78766247-a780f580-79a6-11ea-8c53-f0f04bc0b081.png)

Declare below in component.ts file

`public check : any= {check:'check1',check2:'check2'};`

and declare like below 
` <e-column field="displayValue" headerText="{{check.check}}" width="80"></e-column>
`

it return my header text vale like below
![image](https://user-images.githubusercontent.com/36756050/78766815-60473480-79a7-11ea-8bce-94523d3391cc.png)

**Comment by kumaresan-subramani** [comment]
@alan-agius4 , any idea about this issue?

**Comment by kumaresan-subramani** [comment]
For example i have changed my component declaratin like below

```
<ejs-grid #grid [dataSource]="items" allowPaging="true" allowSorting="true" allowMultiSorting="true" (dataStateChange)="dataStateChange($event)" [locale]="locale" [pageSettings]="pageSettings">
    <e-columns>
      <e-column field="displayValue" headerText="{{check.check}}" width="80"></e-column>
      <e-column field="numberValue" headerText="{{'GRID.NUMBERVALUE' | translate}}" width="75" [allowSorting]="false"></e-column>
      <e-column field="dateValue" headerText="{{'GRID.DATEVALUE' | translate}}" width="75" format="short"></e-column>
      <e-column width="30" headerText="" textAlign="center">
        <ng-template #template let-data>  
          <a (click)="delete(data.keyValue)">Delete</a>
        </ng-template>
      </e-column>
    </e-columns>
  </ejs-grid>
```

### Flow of my issue:

1. Getting 1st column template reference
![image](https://user-images.githubusercontent.com/36756050/78775396-c6d24f80-79b3-11ea-9282-4dc95186cdc8.png)

2. For first column directive reference
![image](https://user-images.githubusercontent.com/36756050/78776281-38f76400-79b5-11ea-83f7-f641c14fe556.png)

3. 2nd column template referecne
![image](https://user-images.githubusercontent.com/36756050/78776430-79ef7880-79b5-11ea-803b-a060e83304bd.png)

4. For 2nd column directive reference
 i have used translate so that `header-text` property is null, refer below image

![image](https://user-images.githubusercontent.com/36756050/78776814-14e85280-79b6-11ea-87e6-370999fab9d5.png)

**Comment by AndrewKushnir** [comment]
Hi @kumaresan-subramani, thanks for reporting the issue.

Could you please help create a minimum possible repro, so that it's easier to debug? Ideally it should be a couple components with simple templates that contain only the content needed to repro the issue. Once the repro is ready, pleased create a Github repo with the demo app and we'll perform additional investigation.

Thank you.

**Comment by kumaresan-subramani** [comment]
HI @AndrewKushnir ,

Thanks for your polite reply,

Please check below sample:

https://github.com/kumaresan-subramani/ngx-translate 

while serve above sample, you will see below in browser page which dont have header-text for my 2nd column [i have used translate for 2nd column]
![image](https://user-images.githubusercontent.com/36756050/78861835-7a3e5100-7a53-11ea-92d4-1f7c23f10565.png)

**Comment by sod** [comment]
Maybe it works if you change

```html
<e-column field="displayValue" headerText="{{'GRID.DISPLAYVALUE' | translate}}" width="80"></e-column>
```

to

```html
<e-column field="displayValue" [headerText]="'GRID.DISPLAYVALUE' | translate" width="80"></e-column>
```

**Comment by kumaresan-subramani** [comment]
Hi @sod ,

Thanks for your reply, this doesn't resolve my issue

**Comment by AndrewKushnir** [comment]
Hi @kumaresan-subramani,

Thanks for providing a repository, I was able to reproduce the issue.

I've performed additional investigation and it looks like the problem is related to the `ngx-translate` and `syncfusion/ej2-angular-ui-components` libs compatibility with Angular Ivy. These libs seem to work correctly when used in isolation, but fail when used together. I've tested the following scenarios:

```
<div title="{{ 'GRID.NUMBERVALUE' | translate }}">Some text</div>
```
and also using <e-column> with standard `json` pipe (just to test the behavior):
```
...
<e-column field="CustomerID" headerText="{{ 'GRID.NUMBERVALUE' | json }}" width="75" [allowSorting]="false"></e-column>
...
```

In both cases the app was rendered correctly and the framework worked as expected.

Maybe try to create a ticket in the `syncfusion/ej2-angular-ui-components` repo to see if they've seen similar issues before and if they can help investigate it further.

Thank you.

**Comment by kumaresan-subramani** [comment]
HI @AndrewKushnir ,

Thanks for validating reported issue.

> Maybe try to create a ticket in the syncfusion/ej2-angular-ui-components repo to see if they've seen similar issues before and if they can help investigate it further.

since this sample working fine in angular 8 without any issue so that i have confirmed that issue does not occurs due to `ngx-translate` and `syncfusion/ej2-angular-ui-components` libs.

Can you please check on this?

**Comment by petebacondarwin** [comment]
I debugged into the reproduction app, and I can see that Angular is setting the `headerText` input value on the `e-column` component instance. But for some reason this is not being rendered... Perhaps the `e-column` component is not realising that its input has changed? Or perhaps the change detection is not being triggered which means that the new value is not being rendered?

**Comment by kumaresan-subramani** [comment]
HI @petebacondarwin ,

Thanks for your reply,

https://github.com/angular/angular/issues/36503#issuecomment-610892113

From my above comment, i have mentioned when angular try to get `e-column` component at initial stage, `headertext` property is null so that i wont know how to handle this at our end.

But same works exactlyfine in angular 8.

I have changed something recently i my source level.refer below

**Comment by petebacondarwin** [comment]
Looking some more...

**Comment by petebacondarwin** [comment]
I can show that the `ngx-translate` package is not the culprit... Here is a minimal reproduction that only uses the `async` pipe.  https://github.com/petebacondarwin/issue-36503

I think the problem appears to be that when the asynchronous value changes, the syncfusion grid (or perhaps just the column directive) is not aware that the change has happened.

**Comment by kumaresan-subramani** [comment]
@petebacondarwin ,

Fro your validation details, i have tried to avoid that time delay using settimeout method in below

https://github.com/syncfusion/ej2-angular-ui-components/blob/f66c91ca4278c7785fb3e4806921f095358278a7/components/grids/src/grid/grid.component.ts#L253

But same issue occurs, i think this wont be handle at syncfusion grid level

**Comment by petebacondarwin** [comment]
It is not clear to me, yet, whether this is a bug in the core Angular change detection or an issue that can be resolved within the ej2-angular-iu-component package. I think we need someone with deep ivy knowledge like @pkozlowski-opensource or @mhevery to take a look...

**Comment by pkozlowski-opensource** [comment]
I'm more than happy to have a look but I would need a reduced reproduction scenario - currently there is a lot going on in the repro (lots of code, 3rd parties). @kumaresan-subramani can you try to distill the essence of the problem to the bare minimum, ideally without using ej2-angular-iu-component package (create a directive that mimics the essential behaviour of this package).

**Comment by kumaresan-subramani** [comment]
HI @pkozlowski-opensource ,

Thanks for your reply.

you can get simple sample in [here](https://github.com/angular/angular/issues/36503#issuecomment-611341655) since i have already simplified my sample. 

> @kumaresan-subramani can you try to distill the essence of the problem to the bare minimum, ideally without using ej2-angular-iu-component package (create a directive that mimics the essential behaviour of this package).

In this sample, i have just rendered grid with ngx-translate. can you please check with this.

Sorry if i did anything wrong.

**my validations** - https://github.com/angular/angular/issues/36503#issuecomment-612731576

**Comment by petebacondarwin** [comment]
@kumaresan-subramani - what @pkozlowski-opensource  is asking for is something much more minimal than that. My reproduction https://github.com/petebacondarwin/issue-36503 is somewhat simpler because it removes the dependency on ngx-translate.

But what we really could do with is a reproduction that had no 3rd party dependencies at all. Instead just had a simple set of self-contained componnets/directives that show the same problem.

Perhaps this could be done by moving the ej2-grid and related components/directives into the reproduction project, removing the syncfusion dependency altogether and the progressively removing/simplifying the code in the project until there is the minimal needed to reproduce the problem.

**Comment by kumaresan-subramani** [comment]
Okay @petebacondarwin , i will share sample without any 3rd party libraries.

**Comment by petebacondarwin** [comment]
❤️

**Comment by kara** [comment]
@kumaresan-subramani Thanks for offering to share a smaller reproduction! Since this issue is not yet actionable and starting to get stale, go ahead and open a new issue when the repro is available and link back to this one. It will bring the issue back to our attention, and at that point, we'll have the information we need to keep investigating.
